src/main/java/com/amazon/opendistroforelasticsearch/security/auditlog/impl/AbstractAuditLog.java

```diff
@@ -80,33 +79,15 @@
 import com.google.common.io.BaseEncoding;
 
 public abstract class AbstractAuditLog implements AuditLog {
+    protected final Logger log = LogManager.getLogger(this.getClass());
 
-    private static final List<String> DEFAULT_DISABLED_CATEGORIES = Arrays.asList(AuditCategory.AUTHENTICATED.toString(), AuditCategory.GRANTED_PRIVILEGES.toString());
+    private final ThreadPool threadPool;
+    private final IndexNameExpressionResolver resolver;
+    private final ClusterService clusterService;
+    private final Settings settings;
+    private final AuditConfig auditConfig;
```

Comment:
If `auditConfig` is `final` and immutable (all fields are also final) how it can be reloaded dynamically?

Reply:
Yes this needs to be non-final when reloaded dynamically. I thought of doing it in #319 when the change is actually getting merged. Till then we don't need it. I feel having rigidity helps in making refactors less error-prone. Thats why I put this as final. What do you think?